src/rank_llm/retrieve/retriever.py

```diff
@@ -141,29 +180,33 @@ def retrieve(
             candidates_file = Path(
                 f"{retrieve_results_dirname}/{self._retrieval_method.name}/retrieve_results_{self._dataset}_top{k}.jsonl"
             )
-            query_name = f"{self._retrieval_method.name}/retrieve_results_{self._dataset}_top{k}.jsonl"
+            max_k_file = self._get_file_with_highest_k(
+                retrieve_results_dirname, self._retrieval_method.name, self._dataset
+            )
             if not candidates_file.is_file():
-                try:
-                    file_path = download_cached_hits(query_name)
-                    with open(file_path, "r") as f:
-                        retrieved_results = []
-                        for line in f:
-                            retrieved_results.append(
-                                from_dict(data_class=Request, data=json.loads(line))
-                            )
-                except ValueError as e:
+                if max_k_file is not None:
+                    print(f"Reusing existing file: {max_k_file} for top {k} reranking.")
+
+                    with open(max_k_file, "r") as f:
+                        retrieved_results = [
+                            from_dict(data_class=Request, data=json.loads(line))
+                            for i, line in enumerate(f)
+                            if i < k
+                        ]
+                else:
                     try:
-                        assert k <= 100
-                        query_name = f"{self._retrieval_method.name}/retrieve_results_{self._dataset}_top100.jsonl"
+                        query_name = f"{self._retrieval_method.name}/retrieve_results_{self._dataset}_top{k}.jsonl"

```

Comment:
The query for caching should not be based on any k.
We should have predefined ks for caching (maybe 100 and 1000) and use them for downloading the cached file based on the k range. if k<=100 query would be ..._100.jsonl, if 100<k<=1000 query would be ..._1000.jsonl
@ronakice does this seem good?